transport/grpc/handler.go

```diff
@@ -97,6 +113,32 @@ func getServiceAndProcedureFromMethod(streamMethod string) (string, string, erro
 	return service, procedure, nil
 }
 
+func getContextMetadata(ctx context.Context) (metadata.MD, error) {
+	contextMetadata, ok := metadata.FromContext(ctx)
+	if !ok || contextMetadata == nil {
+		return nil, errCantExtractMetadata{ctx: ctx}
+	}
+	return contextMetadata, nil
+}
+
+func getCaller(ctxMetadata metadata.MD) (string, error) {
+	// TODO Make a gRPC outbounds require a header for caller and enforce it at inbounds
+	return extractMetadataHeader(ctxMetadata, CallerHeader)
+}
+
+func getEncoding(ctxMetadata metadata.MD) (string, error) {
```

Comment:
I would get rid of the `getCaller` and `getEncoding` functions. The call to
`extractMetadataHeader` is obvious enough with `CallerHeader`/`EncodingHeader`.